Update dockerfiles to run locally

elliottower opened the pull request:
# Description

This allows users to run the same dockerfiles which we use for CI testing locally, without having to modify the code manually. The PYTHON_VERSION arg is set to default values for each environment in case the user doesn't supply the arg, and it checks if there is a pyproject.toml file in the /usr/local/shimmy/ directory and installs from there if it does exist, otherwise installs via `pip install git+[repo]` to fetch the latest version. We could change that to fetch the latest release instead though if we want things to be a bit more sectioned off (this command should work `pip install "git+https://github.com/USER/REPO@$(curl -s https://api.github.com/repos/USER/REPO/releases/latest | jq -r ".tag_name")"`, from https://stackoverflow.com/questions/68315468/pip-install-latest-release-from-github)

The dockerfiles are very handy for installing DM lab and meltingpot because they run all the shell scripts for you and make everything neat and automatic, and also these environments do not support mac (meltingpot does but the install script is glitched) or windows. These could be useful for the future for things like deploying to kubernetes or running training in the cloud, I think it's just overall a good thing to have dockerfiles provided for each environment which can be used outside of our narrow use case (which didn't require much editing to get working)

I cleaned up the meltingpot code and did some other stylistic changes to make all the dockerfiles match as well.

Fixes # (issue), Depends on # (pull request)

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

### Screenshots

> Please attach before and after screenshots of the change if applicable.
> To upload images to a PR -- simply drag and drop or copy paste.

# Checklist:

- [ ] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [ ] I have run `pytest -v` and no errors are present.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

<!--
You can mark something as done by putting an x character in it

For example,
- [x] I have done this task
- [ ] I have not done this task
-->

pseudo-rnd-thoughts reviewed (changes requested):
I just looked at the CI and realised that the tests are finishing in 1 second which is meaning that they are not running

elliottower commented:
> I just looked at the CI and realised that the tests are finishing in 1 second which is meaning that they are not running

This was an issue with the docker entrypoint, I had changed the exec line to run bash as that was what got local docker instances to work, but I realized I can just keep the same exec and do docker run bash, so I updated the docs to say that and reverted the change.
